Automattic/wp-calypso
# Eliminate the setCurrentUserFlags action and lib/redux-helpers

Exploits a cleanup opportunity I discovered when working on the boot sequence of the Sidebar mini-app experiment (#31120):

The `CURRENT_USER_FLAGS_RECEIVE` is redundant: we can set the `state.currentUser.flags` array just fine when processing the `CURRENT_USER_RECEIVE` action.

Merging them together eliminates the `lib/redux-helpers` code that provided a helper to dispatch both actions by one function call. And nothing else.

**How to test:**
Check that the current user in Calypso boots as expected. Some server code runs only when user bootstrapping is active. That will need to be carefully tested on staging while deploying.

Test that features that use the user flags work correctly. Currently the only such feature are domains and the `DOMAINS_WITH_PLANS_ONLY` flag. Cc: @Automattic/cobalt
